### PR TITLE
Null replication in synthetic data generation

### PR DESCRIPTION
**Pull request by Ta7ar**
This PR incorporates null values into synthetic data generation to mimic null values from original dataset. 

To generate null values, the `DataProfiler` report needs the field `null_replication_metrics` (enabled by options `"null_replication_metrics.is_enabled": True`). `DataProfiler` version `0.7.9` does not support that. Therefore to run test demo script the latest version of `DataProfiler` (higher than `0.7.9`) needs to be installed. 

`synthetic_data.py` will work as usual if no `null_replication_metrics` is present in the report.

_Todo_:
* Update `DataProfiler` version in requirements once version with `null_replication_metrics` is released
* Write test cases for `null_replication.py`

### CODE REVIEW

**Comment by Ta7ar** [comment]
Changes in latest commit:
* Update `DataProfiler` requirements from `0.7.9` to `0.7.10`
* Added test cases for null replication

File: synthetic_data/null_replication.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        for c in [0, 1]:
+            prior = np.log(priors[c])
+            inv_cov = np.linalg.inv(cov)
+            diff = x - means[c]
+            likelihood = -0.5 * diff.T @ inv_cov @ diff
+            posterior = prior + likelihood
+            posts.append(posterior)
+        pred = np.argmax(posts)
+        preds.append(pred)
+    return np.array(preds)
```
Comment (taylorfturner) [inline synthetic_data/null_replication.py:60]:
```suggestion
    for iter_x in X:
        posts = list()
        for c in [0, 1]:
            prior = np.log(priors[c])
            inv_cov = np.linalg.inv(cov)
            diff = iter_x - means[c]
            likelihood = -0.5 * diff.T @ inv_cov @ diff
            posterior = prior + likelihood
            posts.append(posterior)
        pred = np.argmax(posts)
        preds.append(pred)
    return np.array(preds)
```

**Review by taylorfturner — APPROVED** [review]
couple things I would change here, which I have added comments for, but will approve for now. we can do a subsequent PR to tiddy up a bit

**Review by tazitoo — APPROVED** [review]
LGTM

**Pull request merged**